### PR TITLE
Add Astro.generator

### PR DESCRIPTION
**Pull request by natemoo-re**
## Changes

- Adds `Astro.generator` which returns `Astro v1.0.0-beta.x` as a string.
- Includes the following in all examples

```astro
<meta name="generator" content={Astro.generator} />
```

- Mostly just to make using this more convenient, it's come up quite a few times on Discord.

## Testing

Tests added

## Docs

https://github.com/withastro/docs/pull/1231

### CODE REVIEW

**Comment by sacarvy** [comment]
Yep, this is good but I am thinking most Useful Component that can Astro can provide is `<SEO />` build into the Astro, because it's very hard to write whole bunch of meta tags title tags as well open-graph, or instead of installing a third party library, 
A static Site Generator can have a build-in `SEO` component, It's totally optional for users and It's very useful for MPA and Portfolios.

**Comment by natemoo-re** [comment]
> Yep, this is good but I am thinking most Useful Component that can Astro can provide is `<SEO />` build into the Astro, because it's very hard to write whole bunch of meta tags title tags as well open-graph, or instead of installing a third party library, A static Site Generator can have a build-in `SEO` component, It's totally optional for users and It's very useful for MPA and Portfolios.

I agree! I know there are some community members working on this, would be very nice to pull into an official `@astrojs/seo` or something like that.

**Comment by JulianCataldo** [comment]
SEO meta tags are quite a magic sauce, for example when I used Nuxt/Next, I didn't use SEO components because it added a lot of superfluous thing for me… and some bits were missing.
Was just simpler to craft it from scratch (not complicated, and you are in total control).
Just having a component for "generator" might encourage those who don't want a full-fledge SEO component to add astro reference (in the end, it's good for the community as a whole, like for stats…).
Just thinking aloud😄

File: packages/astro/components/Generator.astro [hunk 1]
```diff
@@ -0,0 +1 @@
+<meta name="generator" content={Astro.generator} />
```
Comment (FredKSchott) [inline packages/astro/components/Generator.astro:1]: in the interest of a smaller core, is `Astro.generator` enough to accomplish this goal? The dedicated component feels like overkill, and when I hear "Generator component" I immediately thought of some advanced Astro concept. 

Having this snippet example available somewhere in our docs feels like 90% of the value!

Comment (natemoo-re) [reply packages/astro/components/Generator.astro:1]: Agreed! I'm in favor of only adding `Astro.generator` and putting this snippet in our docs and examples.

Comment (natemoo-re) [reply packages/astro/components/Generator.astro]: Removed `Generator` in [a07f316](https://github.com/withastro/astro/pull/4012/commits/a07f31673982ebd0812f532fccfc8b406937e674)

**Comment by matthewp** [comment]
Looks like this is breaking some tests.

**Review by matthewp — APPROVED** [review]

**Pull request merged**